src/samui/backend/samui/api/oauth.py

```diff
@@ -0,0 +1,50 @@
+from base64 import b64encode
+import os
+from urllib.parse import urlencode
+
+import requests
+
+from flask import Blueprint, request, jsonify
+import logbook
+
+logger = logbook.Logger(__name__)
+
+oauth_api = Blueprint('oauth', __name__)
+
+OAUTH_CLIENT_ID = os.environ.get('OAUTH_CLIENT_ID', '')
+OAUTH_SECRET_ID = os.environ.get('OAUTH_SECRET_ID', '')
+
+
+@oauth_api.route('/redirect', methods=['POST'])
+def oauth_redirect():
+    json = request.get_json()
+    account = json.get('account')
+    return_href = json.get('return_href')
+
+    return jsonify(url=f"https://{account}.snowflakecomputing.com/oauth/authorize?" + urlencode({
+        'client_id': OAUTH_CLIENT_ID,
+        'response_type': 'code',
+        'scope': 'refresh_token',
+        'redirect_uri': return_href,
+    }))
+
+
+@oauth_api.route('/return', methods=['POST'])
+def oauth_return():
+    json = request.get_json()
+    code = json.get('code')
+    account = json.get('account')
+
+    response = requests.post(
+        f'https://{account}.snowflakecomputing.com/oauth/token-request',
+        headers={
+            'Content-Type': 'application/x-www-form-urlencoded',
+            'Authorization': 'Basic ' + b64encode(f"{OAUTH_CLIENT_ID}:{OAUTH_SECRET_ID}".encode('utf8')).decode('utf8'),
```

Comment:
Oauth requires that you use basic auth to encode the secret id in the request?

Reply:
talked about this offline, decided it's fine

Reply:
yeah -- I think this specific piece is part of OAuth standard